ivy/functional/frontends/torch/nn/functional/vision_functions.py

```diff
@@ -355,3 +355,272 @@ def upsample_bilinear(input, size=None, scale_factor=None):
 @to_ivy_arrays_and_back
 def upsample_nearest(input, size=None, scale_factor=None):
     return interpolate(input, size=size, scale_factor=scale_factor, mode="nearest")
+
+
+def reflect(x, low2, high2):
+    min = low2 / 2
+    span = (high2 - low2) / 2
+    x = ivy.abs(x - min)
+    frac_in = ivy.abs(x / span)
+    extra = (frac_in - ivy.floor(frac_in)) * ivy.abs(span)
+    flips = ivy.floor(x / span)
+    x[flips % 2 == 0] = (extra + min)[flips % 2 == 0]
+    x[flips % 2 != 0] = (span - extra + min)[flips % 2 != 0]
+    return x
+
+
+def bicubic_interp(x, t, alpha=-0.75):
+    n, h, w = t.shape
+    coeffs = []
+    coeffs.append(ivy.reshape(cubic_conv2(alpha, t + 1), (n, 1, h, w)))
+    coeffs.append(ivy.reshape(cubic_conv1(alpha, t), (n, 1, h, w)))
+    coeffs.append(ivy.reshape(cubic_conv1(alpha, 1 - t), (n, 1, h, w)))
+    coeffs.append(ivy.reshape(cubic_conv2(alpha, 2 - t), (n, 1, h, w)))
+    return x[0] * coeffs[0] + x[1] * coeffs[1] + x[2] * coeffs[2] + x[3] * coeffs[3]
+
+
+def grid_sample_padding(grid, padding_mode, align_corners, borders=None):
+    if padding_mode == "reflection":
+        if align_corners:
+            for idx, border in enumerate(borders):
+                grid[..., idx] = reflect(grid[..., idx], 0, 2 * (border - 1))
+                grid[..., idx] = ivy.clip(grid[..., idx], 0, border - 1)
+
+        else:
+            for idx, border in enumerate(borders):
+                grid[..., idx] = reflect(grid[..., idx], -1, 2 * border - 1)
+                grid[..., idx] = ivy.clip(grid[..., idx], 0, border - 1)
+
+    elif padding_mode == "border":
+        for idx, border in enumerate(borders):
+            grid[..., idx] = ivy.clip(grid[..., idx], 0, border - 1)
+
+    masks = []
+    for idx, border in enumerate(borders):
+        masks.append(ivy.bitwise_or(grid[..., idx] < -4, grid[..., idx] > border + 2))
+        borders[idx] += 1
+
+    zeros_mask = masks[0]
+    for i in range(1, len(borders)):
+        zeros_mask = ivy.bitwise_or(zeros_mask, masks[i])
+
+    if grid[zeros_mask].shape[0] > 0:
+        grid[zeros_mask] = ivy.array(borders)
+    return grid
+
+
+cubic_conv1 = lambda A, x: ((A + 2) * x - (A + 3)) * x * x + 1
+cubic_conv2 = lambda A, x: (((A * x) - (5 * A)) * x + (8 * A)) * x - (4 * A)
+
+
+@with_supported_dtypes({"2.0.1 and below": ("float32", )}, "torch")
```

Comment:
```suggestion
@with_supported_dtypes({"2.0.1 and below": ("float32",  "float64")}, "torch")
```
float64 is still a valid type. I've checked and the tests still pass with float64 included